Index: rtl8821cs-wifi/core/rtw_ap.c
===================================================================
--- rtl8821cs-wifi.orig/core/rtw_ap.c
+++ rtl8821cs-wifi/core/rtw_ap.c
@@ -131,8 +131,11 @@ static void update_BCNTIM(_adapter *pada
 			if (p !=  NULL)
 				offset += tmp_len + 2;
 
-			/*DS Parameter Set IE, len=3*/
-			offset += 3;
+			/* DS Parameter Set IE */
+			p = rtw_get_ie(pie + _BEACON_IE_OFFSET_, _DSSET_IE_, &tmp_len,
+						(pnetwork_mlmeext->IELength - _BEACON_IE_OFFSET_));
+			if (p != NULL)
+				offset += tmp_len + 2;
 
 			premainder_ie = pie + offset;
 
